plugins/vpp/l3plugin/vppcalls/vpp1908/route_dump.go

```diff
@@ -25,85 +24,90 @@ import (
-		stop, err := reqCtx.ReceiveReply(fibDetails)
-		if stop {
-			break
-		}
+func (h *RouteHandler) DumpRoutes() (routes []*vppcalls.RouteDetails, err error) {
+	// dump routes for every VRF and for both IP versions
+	for _, vrfID := range h.vrfIndexes.ListAllVrfIDs() {
+		// IPv6
+		ipv6Routes, err := h.dumpRoutesForVrfAndIP(vrfID, true)
 		if err != nil {
 			return nil, err
 		}
-		ipv4Route, err := h.dumpRouteIPv4Details(fibDetails)
+		routes = append(routes, ipv6Routes...)
+		// IPv4
+		ipv4Routes, err := h.dumpRoutesForVrfAndIP(vrfID, false)
 		if err != nil {
 			return nil, err
 		}
-		routes = append(routes, ipv4Route...)
+		routes = append(routes, ipv4Routes...)
+
 	}
+	return routes, nil
+}
 
-	// Dump IPv6 l3 FIB.
-	reqCtx = h.callsChannel.SendMultiRequest(&l3binapi.IP6FibDump{})
+// dumpRoutesForVrf returns routes for given VRF and IP versiob
+func (h *RouteHandler) dumpRoutesForVrfAndIP(vrfID uint32, isIPv6 bool) (routes []*vppcalls.RouteDetails, err error) {
+	reqCtx := h.callsChannel.SendMultiRequest(&l3binapi.IPRouteDump{
+		Table: l3binapi.IPTable{
+			TableID: vrfID,
+			IsIP6:   boolToUint(isIPv6),
+		},
+	})
 	for {
-		fibDetails := &l3binapi.IP6FibDetails{}
+		fibDetails := &l3binapi.IPRouteDetails{}
 		stop, err := reqCtx.ReceiveReply(fibDetails)
 		if stop {
 			break
 		}
 		if err != nil {
-			return nil, err
+			return routes, err
 		}
-		ipv6Route, err := h.dumpRouteIPv6Details(fibDetails)
+		ipRoute, err := h.dumpRouteIPDetails(fibDetails.Route)
 		if err != nil {
-			return nil, err
+			return routes, err
 		}
-		routes = append(routes, ipv6Route...)
+		routes = append(routes, ipRoute...)
 	}
 
 	return routes, nil
 }
 
-func (h *RouteHandler) dumpRouteIPv4Details(fibDetails *l3binapi.IPFibDetails) ([]*vppcalls.RouteDetails, error) {
-	return h.dumpRouteIPDetails(fibDetails.TableID, fibDetails.TableName, fibDetails.Address, fibDetails.AddressLength, fibDetails.Path, false)
-}
-
-func (h *RouteHandler) dumpRouteIPv6Details(fibDetails *l3binapi.IP6FibDetails) ([]*vppcalls.RouteDetails, error) {
-	return h.dumpRouteIPDetails(fibDetails.TableID, fibDetails.TableName, fibDetails.Address, fibDetails.AddressLength, fibDetails.Path, true)
-}
-
 // dumpRouteIPDetails processes static route details and returns a route objects. Number of routes returned
 // depends on size of path list.
-func (h *RouteHandler) dumpRouteIPDetails(tableID uint32, tableName []byte, address []byte, prefixLen uint8, paths []l3binapi.FibPath, ipv6 bool) ([]*vppcalls.RouteDetails, error) {
+func (h *RouteHandler) dumpRouteIPDetails(ipRoute l3binapi.IPRoute) ([]*vppcalls.RouteDetails, error) {
 	// Common fields for every route path (destination IP, VRF)
 	var dstIP string
-	if ipv6 {
-		dstIP = fmt.Sprintf("%s/%d", net.IP(address).To16().String(), uint32(prefixLen))
+	netIP := make([]byte, 16)
+	if ipRoute.Prefix.Address.Af == l3binapi.ADDRESS_IP6 {
+		ip6Addr := ipRoute.Prefix.Address.Un.GetIP6()
+		copy(netIP[:], ip6Addr[:])
```

Comment:
There is no need for extra copying here. We only use the `ip6Addr` for conversion to IP string.

```go
ip6Addr := d.Punt.Nh.Un.GetIP6()
dstIP = net.IP(ip6Addr [:]).To16().String()
```